Assets/Scripts/Game/DaggerfallExteriorAutomap.cs

```diff
@@ -762,9 +763,34 @@ private void createBuildingNameplates(DFLocation location)
                             PlayerGPS.DiscoveredBuilding discoveredBuilding;
                             if (GameManager.Instance.PlayerGPS.GetDiscoveredBuilding(buildingSummary.buildingKey, out discoveredBuilding))
                             {
-                                // only create nameplates for buildings different than "Residence"
-                                if (discoveredBuilding.displayName != "Residence")
+                                
+                                if (!RMBLayout.IsResidence(buildingSummary.BuildingType)) // guilds, shops, taverns, palace (general case)
+                                {
                                     newBuildingNameplate.name = discoveredBuilding.displayName;
+                                }
+                                else if (RMBLayout.IsResidence(buildingSummary.BuildingType)) // residence handling
+                                {
+                                    // check if residence is involved in active quest (this might be replaced by function call - for now I implemented this by myself
```

Comment:
I'll roll this into a nice dictionary lookup in QuestMachine that gets refreshed when quests are compiled or removed. That way the automap population only need to a do a single lookup against the dict. This should work great for now though! :)